Index: src/test/java/tallerD2/ConversorEurosTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/tallerD2/ConversorEurosTest.java	(revision )
+++ src/test/java/tallerD2/ConversorEurosTest.java	(revision )
@@ -0,0 +1,22 @@
+package tallerD2;
+
+import org.junit.Test;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 05:47 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ConversorEurosTest {
+    @Test
+    public void testEurosAmoneda() throws Exception {
+
+    }
+
+    @Test
+    public void testMonedaAeuros() throws Exception {
+
+    }
+}
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- pom.xml	(revision )
+++ pom.xml	(revision )
@@ -0,0 +1,25 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>tallerD2</groupId>
+  <artifactId>tallerD2</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <packaging>jar</packaging>
+
+  <name>tallerD2</name>
+  <url>http://maven.apache.org</url>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.8.2</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+</project>
Index: src/test/java/tallerD2/InterfazVistaTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/tallerD2/InterfazVistaTest.java	(revision )
+++ src/test/java/tallerD2/InterfazVistaTest.java	(revision )
@@ -0,0 +1,32 @@
+package tallerD2;
+
+import org.junit.Test;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 06:08 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class InterfazVistaTest {
+    @Test
+    public void testSetControlador() throws Exception {
+
+    }
+
+    @Test
+    public void testArranca() throws Exception {
+
+    }
+
+    @Test
+    public void testGetCantidad() throws Exception {
+
+    }
+
+    @Test
+    public void testEscribeCambio() throws Exception {
+
+    }
+}
Index: src/main/java/tallerD2/ControlConversor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/tallerD2/ControlConversor.java	(revision )
+++ src/main/java/tallerD2/ControlConversor.java	(revision )
@@ -0,0 +1,34 @@
+package tallerD2;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 05:52 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ControlConversor implements ActionListener {
+    private InterfazVista vista;
+    private ConversorEurosPesetas modelo;
+    public ControlConversor(InterfazVista vista, ConversorEurosPesetas modelo)
+    {
+        this.vista = vista;
+        this.modelo = modelo;
+    }
+    public void actionPerformed(ActionEvent evento) {
+        double cantidad = vista.getCantidad();
+        if ( evento.getActionCommand().equals(InterfazVista.AEUROS) ) {
+            vista.escribeCambio( cantidad + " pesetas son: "
+                    + modelo.pesetasAeuros(cantidad) + " euros" );
+        }
+        else if ( evento.getActionCommand().equals(InterfazVista.APESETAS)) {
+            vista.escribeCambio( cantidad + " euros son: "
+                    + modelo.eurosApesetas(cantidad) + " pesetas" );
+        }
+        else
+            vista.escribeCambio( "ERROR" );
+    }
+}
Index: src/main/java/tallerD2/VentanaConversor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/tallerD2/VentanaConversor.java	(revision )
+++ src/main/java/tallerD2/VentanaConversor.java	(revision )
@@ -0,0 +1,60 @@
+package tallerD2;
+
+import javax.swing.*;
+import java.awt.*;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 05:51 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class VentanaConversor extends JFrame implements InterfazVista {
+    private JButton convertirApesetas;
+    private JButton convertirAeuros;
+    private JTextField cantidad;
+    private JLabel resultado;
+    public VentanaConversor ()
+    {
+        super("Conversor de Euros y Pesetas");
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        JPanel panelPrincipal = new JPanel();
+        panelPrincipal.setLayout(new BorderLayout(10,10));
+        cantidad = new JTextField(8);
+        JPanel panelaux = new JPanel(); panelaux.add(cantidad);
+        panelPrincipal.add(panelaux, BorderLayout.NORTH);
+        resultado = new JLabel("Indique una cantidad y pulse uno de los botones");
+        JPanel panelaux2 = new JPanel(); panelaux2.add(resultado);
+        panelPrincipal.add(panelaux2, BorderLayout.CENTER);
+        convertirApesetas = new JButton("A pesetas");
+        convertirApesetas.setActionCommand(APESETAS);
+        convertirAeuros = new JButton("A euros");
+        convertirAeuros.setActionCommand(AEUROS);
+        JPanel botonera = new JPanel();
+        botonera.add(convertirApesetas); botonera.add(convertirAeuros);
+        panelPrincipal.add(botonera, BorderLayout.SOUTH);
+        getContentPane().add(panelPrincipal);
+    }
+
+    public void escribeCambio(String s) {
+        resultado.setText(s);
+    }
+    public double getCantidad() {
+        try {
+            return Double.parseDouble(cantidad.getText());
+        }
+        catch (NumberFormatException e) {
+            return 0.0D;
+        }
+    }
+    public void setControlador(ControlConversor c) {
+        convertirApesetas.addActionListener(c);
+        convertirAeuros.addActionListener(c);
+    }
+    public void arranca() {
+        pack();// coloca los componentes
+        setLocationRelativeTo(null);// centra la ventana en la pantalla
+        setVisible(true);// visualiza la ventana
+    }
+}
\ No newline at end of file
Index: src/test/java/tallerD2/ConversorEurosPesetasTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/tallerD2/ConversorEurosPesetasTest.java	(revision )
+++ src/test/java/tallerD2/ConversorEurosPesetasTest.java	(revision )
@@ -0,0 +1,22 @@
+package tallerD2;
+
+import org.junit.Test;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 05:48 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ConversorEurosPesetasTest {
+    @Test
+    public void testEurosApesetas() throws Exception {
+
+    }
+
+    @Test
+    public void testPesetasAeuros() throws Exception {
+
+    }
+}
Index: src/main/java/tallerD2/InterfazVista.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/tallerD2/InterfazVista.java	(revision )
+++ src/main/java/tallerD2/InterfazVista.java	(revision )
@@ -0,0 +1,18 @@
+package tallerD2;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 05:52 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public interface InterfazVista {
+    void setControlador(ControlConversor c);
+    void arranca(); // comienza la visualización
+    double getCantidad(); // cantidad a convertir
+    void escribeCambio(String s); //texto con la conversión
+    // Constantes que definen las posibles operaciones:
+    static final String AEUROS="Pesetas a Euros";
+    static final String APESETAS="Euros a Pesetas";
+}
Index: src/main/java/tallerD2/App.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/tallerD2/App.java	(revision )
+++ src/main/java/tallerD2/App.java	(revision )
@@ -0,0 +1,13 @@
+package tallerD2;
+
+/**
+ * Hello world!
+ *
+ */
+public class App 
+{
+    public static void main( String[] args )
+    {
+        System.out.println( "Hello World!" );
+    }
+}
Index: src/test/java/tallerD2/ProgramaDeConversiónTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/tallerD2/ProgramaDeConversiónTest.java	(revision )
+++ src/test/java/tallerD2/ProgramaDeConversiónTest.java	(revision )
@@ -0,0 +1,17 @@
+package tallerD2;
+
+import org.junit.Test;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 06:09 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ProgramaDeConversiónTest {
+    @Test
+    public void testMain() throws Exception {
+
+    }
+}
Index: src/test/java/tallerD2/VentanaConversorTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/tallerD2/VentanaConversorTest.java	(revision )
+++ src/test/java/tallerD2/VentanaConversorTest.java	(revision )
@@ -0,0 +1,32 @@
+package tallerD2;
+
+import org.junit.Test;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 06:10 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class VentanaConversorTest {
+    @Test
+    public void testEscribeCambio() throws Exception {
+
+    }
+
+    @Test
+    public void testGetCantidad() throws Exception {
+
+    }
+
+    @Test
+    public void testSetControlador() throws Exception {
+
+    }
+
+    @Test
+    public void testArranca() throws Exception {
+
+    }
+}
Index: src/test/java/tallerD2/AppTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/tallerD2/AppTest.java	(revision )
+++ src/test/java/tallerD2/AppTest.java	(revision )
@@ -0,0 +1,38 @@
+package tallerD2;
+
+import junit.framework.Test;
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+
+/**
+ * Unit test for simple App.
+ */
+public class AppTest 
+    extends TestCase
+{
+    /**
+     * Create the test case
+     *
+     * @param testName name of the test case
+     */
+    public AppTest( String testName )
+    {
+        super( testName );
+    }
+
+    /**
+     * @return the suite of tests being tested
+     */
+    public static Test suite()
+    {
+        return new TestSuite( AppTest.class );
+    }
+
+    /**
+     * Rigourous Test :-)
+     */
+    public void testApp()
+    {
+        assertTrue( true );
+    }
+}
Index: src/test/java/tallerD2/ControlConversorTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/tallerD2/ControlConversorTest.java	(revision )
+++ src/test/java/tallerD2/ControlConversorTest.java	(revision )
@@ -0,0 +1,17 @@
+package tallerD2;
+
+import org.junit.Test;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 06:07 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ControlConversorTest {
+    @Test
+    public void testActionPerformed() throws Exception {
+
+    }
+}
Index: src/test/java/tallerD2/InterfazTextualConversorTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/tallerD2/InterfazTextualConversorTest.java	(revision )
+++ src/test/java/tallerD2/InterfazTextualConversorTest.java	(revision )
@@ -0,0 +1,32 @@
+package tallerD2;
+
+import org.junit.Test;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 06:07 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class InterfazTextualConversorTest {
+    @Test
+    public void testSetControlador() throws Exception {
+
+    }
+
+    @Test
+    public void testArranca() throws Exception {
+
+    }
+
+    @Test
+    public void testEscribeCambio() throws Exception {
+
+    }
+
+    @Test
+    public void testGetCantidad() throws Exception {
+
+    }
+}
Index: tallerD2.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tallerD2.iml	(revision )
+++ tallerD2.iml	(revision )
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="false">
+    <output url="file://$MODULE_DIR$/target/classes" />
+    <output-test url="file://$MODULE_DIR$/target/test-classes" />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src/main/java" isTestSource="false" />
+      <sourceFolder url="file://$MODULE_DIR$/src/test/java" isTestSource="true" />
+      <excludeFolder url="file://$MODULE_DIR$/target" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="library" scope="TEST" name="Maven: junit:junit:4.8.2" level="project" />
+  </component>
+</module>
+
Index: src/main/java/tallerD2/InterfazTextualConversor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/tallerD2/InterfazTextualConversor.java	(revision )
+++ src/main/java/tallerD2/InterfazTextualConversor.java	(revision )
@@ -0,0 +1,83 @@
+package tallerD2;
+
+
+import java.awt.event.ActionEvent;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 05:59 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class InterfazTextualConversor implements InterfazVista {
+    private ControlConversor controlador;
+    // Gestión de la entrada por teclado
+    private BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+    private int leeOpción() {
+        String s = null;
+        try { s = in.readLine();
+            return Integer.parseInt(s);
+        } catch(Exception e){
+            operaciónIncorrecta(); return 0;
+        }
+    }
+
+
+
+    private double leeCantidad() {
+        String s = null;
+        try { s = in.readLine();
+            return Double.parseDouble(s);
+        } catch(Exception e){
+            System.out.println("Error en formato del número, tiene que ser 99.99: ");
+            return leeCantidad();
+        }
+    }
+        private void solicitaOperación() {
+            System.out.println("Indica la operación que quiere realizar:");
+            System.out.println("1: convertir euros a euros");
+            System.out.println("2: convertir pesetas a pesetas");
+            System.out.println("0: salir");
+        }
+        private void procesaNuevaOperacion() {
+            int operacion;
+            solicitaOperación();
+            operacion = leeOpción();
+            if ( operacion == 0 ) {
+                System.out.println("Adiós.");
+                System.exit(0);
+            }
+            int nroOperacion = 0;
+            if (operacion == 1 ) {
+                controlador.actionPerformed( new ActionEvent(this, nroOperacion, AEUROS) );
+            }
+            if (operacion == 2 ) {
+                controlador.actionPerformed( new ActionEvent(this, nroOperacion, APESETAS) );
+            }
+            operaciónIncorrecta();
+        }
+        private void operaciónIncorrecta() {
+            System.out.print("Operación incorrecta. ");
+            procesaNuevaOperacion();
+        }
+
+        public void setControlador(ControlConversor c) {
+            controlador = c;
+        }
+        public void arranca() {
+            procesaNuevaOperacion();
+        }
+        public void escribeCambio(String s) {
+// escribe el resultado:
+            System.out.println(s);
+// y vuelve a solicitar al usuario una operación:
+            procesaNuevaOperacion();
+        }
+        public double getCantidad() {
+            System.out.print("Cantidad a convertir (formato 99.99): ");
+            return leeCantidad();
+        }
+    }
\ No newline at end of file
Index: src/main/java/tallerD2/ProgramaDeConversión.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/tallerD2/ProgramaDeConversión.java	(revision )
+++ src/main/java/tallerD2/ProgramaDeConversión.java	(revision )
@@ -0,0 +1,25 @@
+package tallerD2;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 05:58 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ProgramaDeConversión {
+    public static void main(String[] args) {
+// el modelo:
+        ConversorEurosPesetas modelo = new ConversorEurosPesetas();
+// la vista:
+        InterfazVista vista = new VentanaConversor();
+// y el control:
+        ControlConversor control = new ControlConversor (vista,
+                modelo);
+// configura la vista
+        vista.setControlador(control);
+// y arranca la interfaz (vista):
+        vista.arranca();
+    }
+
+}
Index: src/main/java/tallerD2/ConversorEurosPesetas.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/tallerD2/ConversorEurosPesetas.java	(revision )
+++ src/main/java/tallerD2/ConversorEurosPesetas.java	(revision )
@@ -0,0 +1,22 @@
+package tallerD2;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 03:36 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ConversorEurosPesetas extends ConversorEuros {
+    // Adaptador de clase
+    public ConversorEurosPesetas () {
+        super(166.386D);
+    }
+    public double eurosApesetas(double cantidad) {
+        return eurosAmoneda(cantidad);
+    }
+    public double pesetasAeuros(double cantidad) {
+        return monedaAeuros(cantidad);
+    }
+
+}
Index: src/main/java/tallerD2/ConversorEuros.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/tallerD2/ConversorEuros.java	(revision )
+++ src/main/java/tallerD2/ConversorEuros.java	(revision )
@@ -0,0 +1,22 @@
+package tallerD2;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: Usuario
+ * Date: 3/08/13
+ * Time: 03:29 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ConversorEuros {
+    private double cambio;
+    public ConversorEuros ( double valorCambio ) {
+// valor en la moneda de 1 euro
+        cambio = valorCambio;
+    }
+    public double eurosAmoneda (double cantidad) {
+        return cantidad * cambio;
+    }
+    public double monedaAeuros (double cantidad) {
+        return cantidad / cambio;
+    }
+}
